File: Assignments/Assignment 41/C Codes/program41_3.c
// 3. Write a recursive program which display below pattern.
// count of string (string length)
// Input :  Hello
// Output : 5

#include<stdio.h>

int Display(char *str)
{
    static int iCount = 0;

    if(*str != '\0')
    {
        iCount++;
        str++;
        Display(str);               // recursive call  // confusion here 
    }

    return iCount;
}

int main()
{
    char cValue[] = {"\0"};
    int iRet = 0;

    printf("Enter String : \n");
    scanf("%[^'\n']s", &cValue);

    iRet = Display(cValue);

    printf("%d\n", iRet);
    
    return 0;
}

File: Assignments/Assignment 41/Java Codes/program41_3.java
// 3. Write a recursive program which display below pattern.
// count of string (string length)
// Input :  Hello
// Output : 5

import java.util.Scanner;

class Test
{
    static int iCnt = 0;

    int Display(String str)
    {
        char Arr[] = str.toCharArray();

        if(iCnt < Arr.length)
        {
            iCnt++;
            Display(str);
        }

        return iCnt;
    }
}

class program41_3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        String Arr = null;
        int iRet = 0;

        System.out.println("Enter String : ");
        Arr = sobj.nextLine();

        Test tobj = new Test();
        iRet = tobj.Display(Arr);

        System.out.println(iRet);
    }
}